### PR TITLE
muxer and ingest reporting resilience

### PR DESCRIPTION
**Pull request by kris-watts-gravwell**
This PR does teh following:

1. Removes the global muxer level ingester state reporting routine
2. Adds some functionality in the individual muxer routines to have individual ingestConnections report state
3. Reduces the frequency of ingester state updates when the only thing changing is ingest rates
4. Adds some timeouts to the ingestConnection close routines so that we can detect and close bad connections
5. Adds a new notification boolean on the ingestConnection routine that can signal to the system that grabs new connections that we think the upstream host is unhealthy and we should backoff and sleep before grabbing a new connection

### CODE REVIEW

File: ingest/muxer.go
Comment (john-floren-gravwell) [inline ingest/muxer.go:967]: it'd be nice to finish this comment

File: ingest/muxer.go
Comment (john-floren-gravwell) [inline ingest/muxer.go:949]: we only keep one error here. Might we merge them instead?

**Review by john-floren-gravwell — CHANGES_REQUESTED** [review]

**Comment by ashnwade** [comment]
Would like to get approval from both John and Fritz before merging

File: ingest/muxer.go
Comment (kris-watts-gravwell) [reply ingest/muxer.go]: resolved, fixed the bad comment

File: ingest/muxer.go
Comment (kris-watts-gravwell) [reply ingest/muxer.go]: added merge utility, please validate that i got the logic right.

**Review by david-fritz-gravwell — APPROVED** [review]
LGTM

**Pull request merged**
